db/migrate/20201102105419_change_oda_eligibility_type.rb

```diff
@@ -0,0 +1,10 @@
+class ChangeOdaEligibilityType < ActiveRecord::Migration[6.0]
+  def change
+    add_column :activities, :oda_eligibility_integer, :integer, default: 1
+    Activity.where(oda_eligibility: nil).update_all(oda_eligibility_integer: 1)
+    Activity.where(oda_eligibility: true).update_all(oda_eligibility_integer: 1)
+    Activity.where(oda_eligibility: false).update_all(oda_eligibility_integer: 2)
+    remove_column :activities, :oda_eligibility, :boolean
+    rename_column :activities, :oda_eligibility_integer, :oda_eligibility
+  end
```

Comment:
Although it is annoying the database won't let you do this type change directly, this is exactly the way to go about this sort of change. Sometimes it's done over multiple deploys so that the changes are backward compatible, but I'm not too worried about that in this case.

Reply:
Well, that explains that despite following the documentation and reading several threads on StackOverflow I wasn't able to make it work. Glad to hear this is the way to do it. I was a bit worried this would look too "handcrafted" and someone would point it out (hence, the probably overexplanation on the PR message). 
Regarding the possibility of a rollback, Meyric suggested that probably we wanted this change irreversible to avoid losing any data. For that reason I decided to put everything on the same migration file.

Reply:
That's a good point about reversibility -- Rails won't know how to reverse this operation. We can either mark it as irreversible, or we could implement a `down` method. There's no data loss here, because the new integer values can be mapped back to the old boolean ones. You would have ambiguous data once any records with value `3` are added though.

Reply:
Forgive my new-to-the-project naivety, but could we not put lines 4 - 6 in a data migration, and then remove the  `oda_eligibility` column at a later date? (once this code is confirmed working and deployed to prod) Would keep things a bit cleaner, and would save any data loss if the migrations choke for some reason on deploy.

Reply:
A problem with the `data_migrations` gem (and a reason I'd like to move away from it) is that migrations are not ordered relative to schema migrations. Here, the data changes are part of changing the type of the column so I think they should be part of this migration, not a separate one.

Reply:
"Remove the `oda_eligibility` column at a later date" is what I was getting at with doing this over multiple deploys. Actually doing this kind of change in a fully backward/forward compatible way is a lot more steps and I'm not sure it's worth it in this case.

Reply:
Yeah, I'm not a fan of the `data_migrations` gem either. We moved towards just having the data migrations as plain old Ruby scripts that we just run after deploy

Reply:
As `nil` is being remapped to `1` could the integer field have `null: false` in the migration now?

Reply:
I'll give it a go and see it nothing breaks

Reply:
It worked leeky! 🎉

Reply:
Yeah, this is ace. Much happier with this 👍